### ISSUE TITLE
dumpling error "dumpling.py upload: error: argument --dumppath: expected one argument" on Fedora

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#15472]
https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170125.02/workItem/Common.Tests/details/38037e24-2d39-4dae-a997-6c99c5bcba9d

```
2017-01-25 06:23:16,119: INFO: proc(55): run_and_log_output: Output: echo uploading core to dumpling service
2017-01-25 06:23:16,119: INFO: proc(55): run_and_log_output: Output: python /home/helixbot/.dumpling/dumpling.py upload --dumppath --noprompt --triage full --displayname --properties STRESS_TESTID= --verbose
2017-01-25 06:23:16,119: INFO: proc(55): run_and_log_output: Output: else
2017-01-25 06:23:16,119: INFO: proc(55): run_and_log_output: Output: echo no coredump file was found
2017-01-25 06:23:16,120: INFO: proc(55): run_and_log_output: Output: fi
2017-01-25 06:23:16,120: INFO: proc(55): run_and_log_output: Output: fi
2017-01-25 06:23:16,120: INFO: proc(55): run_and_log_output: Output: return
2017-01-25 06:23:16,120: INFO: proc(55): run_and_log_output: Output: }
2017-01-25 06:23:16,120: INFO: proc(55): run_and_log_output: Output: /home/helixbot/dotnetbuild/work/38037e24-2d39-4dae-a997-6c99c5bcba9d/Payload/dotnet xunit.console.netcore.exe Common.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapp1.1tests -notrait category=OuterLoop -notrait category=failing -notrait category=nonlinuxtests
2017-01-25 06:23:16,120: INFO: proc(55): run_and_log_output: Output: CollectDumps 0 Common.Tests
2017-01-25 06:23:16,120: INFO: proc(55): run_and_log_output: Output: ~/dotnetbuild/work/38037e24-2d39-4dae-a997-6c99c5bcba9d/Work/6d1e3fd1-e72c-4c13-bd58-a884f70b9349/Unzip ~/dotnetbuild/work/38037e24-2d39-4dae-a997-6c99c5bcba9d/Work/6d1e3fd1-e72c-4c13-bd58-a884f70b9349/Unzip
2017-01-25 06:23:16,120: INFO: proc(55): run_and_log_output: Output: executing ulimit -c unlimited
2017-01-25 06:23:16,125: INFO: proc(55): run_and_log_output: Output: Failed to load /home/helixbot/dotnetbuild/work/38037e24-2d39-4dae-a997-6c99c5bcba9d/Payload/shared/Microsoft.NETCore.App/9.9.9/libhostpolicy.so, error: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by /home/helixbot/dotnetbuild/work/38037e24-2d39-4dae-a997-6c99c5bcba9d/Payload/shared/Microsoft.NETCore.App/9.9.9/libhostpolicy.so)
2017-01-25 06:23:16,125: INFO: proc(55): run_and_log_output: Output: An error occurred while loading required library libhostpolicy.so from [/home/helixbot/dotnetbuild/work/38037e24-2d39-4dae-a997-6c99c5bcba9d/Payload/shared/Microsoft.NETCore.App/9.9.9]
2017-01-25 06:23:16,125: INFO: proc(55): run_and_log_output: Output: command exited with ExitCode: 130
2017-01-25 06:23:16,125: INFO: proc(55): run_and_log_output: Output: command failed, trying to collect dumps
2017-01-25 06:23:16,126: INFO: proc(55): run_and_log_output: Output: corefile:
2017-01-25 06:23:16,126: INFO: proc(55): run_and_log_output: Output: uploading core to dumpling service
2017-01-25 06:23:16,214: INFO: proc(55): run_and_log_output: Output: usage: dumpling.py upload [-h] [--verbose] [--squelch] [--noprompt]
2017-01-25 06:23:16,215: INFO: proc(55): run_and_log_output: Output:                           [--logpath LOGPATH] [--url URL]
2017-01-25 06:23:16,215: INFO: proc(55): run_and_log_output: Output:                           [--configpath CONFIGPATH] [--dumppath DUMPPATH]
2017-01-25 06:23:16,215: INFO: proc(55): run_and_log_output: Output:                           [--displayname DISPLAYNAME] [--user USER]
2017-01-25 06:23:16,215: INFO: proc(55): run_and_log_output: Output:                           [--triage {none,client,full}]
2017-01-25 06:23:16,215: INFO: proc(55): run_and_log_output: Output:                           [--incpaths [INCPATHS [INCPATHS ...]]]
2017-01-25 06:23:16,215: INFO: proc(55): run_and_log_output: Output:                           [--properties [key=value [key=value ...]]]
2017-01-25 06:23:16,215: INFO: proc(55): run_and_log_output: Output:                           [--propfile PROPFILE]
2017-01-25 06:23:16,215: INFO: proc(55): run_and_log_output: Output: dumpling.py upload: error: argument --dumppath: expected one argument
2017-01-25 06:23:16,222: INFO: proc(55): run_and_log_output: Output: ~/dotnetbuild/work/38037e24-2d39-4dae-a997-6c99c5bcba9d/Work/6d1e3fd1-e72c-4c13-bd58-a884f70b9349/Unzip
2017-01-25 06:23:16,223: INFO: proc(55): run_and_log_output: Output: Finished running tests. End time=06:23:16. Return value was 130
```

@sepidehMS

**Comment by danmosemsft** [comment]
@MattGal does it look to you that this is the reason all Fedora runs have failed? Or downstream?

**Comment by MattGal** [comment]
@schaabs can hopefully comment.

**Comment by schaabs** [comment]
I don't believe the issue is dumpling setup.  The problem with the upload here is that you're trying to upload a dump without specifying the dump path (presumably because no dump was actually created, although I'd need to look at the script itself to confirm).  The following log output shows that the process exited with an non-zero error code, but corefile is an empty string:

    2017-01-25 06:23:16,125: INFO: proc(55): run_and_log_output: Output: command exited with ExitCode: 130
    2017-01-25 06:23:16,125: INFO: proc(55): run_and_log_output: Output: command failed, trying to collect dumps
    2017-01-25 06:23:16,126: INFO: proc(55): run_and_log_output: Output: corefile:
    2017-01-25 06:23:16,126: INFO: proc(55): run_and_log_output: Output: uploading core to dumpling service

This is why you get the error message, because you're essentially trying to upload a dump which does not exist, i.e. dumppath must be specified and be a valid path:

    dumpling.py upload: error: argument --dumppath: expected one argument

Also I don't believe the error which @MattGal highlighted is actually an error message at all, but rather an error message from trying to run the test.

**Comment by danmosemsft** [comment]
@MattGal who owns the next action to get this up and going?

**Comment by MattGal** [comment]
@sepidehMS turned it on, I'd start with her.  Failing that, if @Chrisboh  wants me to work on this I certainly can.

**Comment by danmosemsft** [comment]
OK so it's not dumpling. @weshaggard any light you can cast on this failure? Perhaps this is the issue:
`2017-01-25 06:23:16,118: INFO: proc(55): run_and_log_output: Output: /home/helixbot/dotnetbuild/work/38037e24-2d39-4dae-a997-6c99c5bcba9d/Work/6d1e3fd1-e72c-4c13-bd58-a884f70b9349/Unzip/RunTests.sh: line 89: echo: write error: Invalid argument`
although I can't find "runtests.sh" in our repos.

**Comment by MattGal** [comment]
Runtests.sh is emitted by https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/GenerateTestExecutionScripts.cs and may be tweaked by override build tools stuff.  @mellinoe  may have more to add.

**Comment by MattGal** [comment]
Line 89 in question is:
echo echo 0x3F > /proc/self/coredump_filter

**Comment by sepidehMS** [comment]
> error: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by /home/helixbot/dotnetbuild/work/38037e24-2d39-4dae-a997-6c99c5bcba9d/Payload/shared/Microsoft.NETCore.App/9.9.9/libhostpolicy.so


As @schaabs explained earlier, this error is not related to Dumpling. And the failure in Fedora runs is coming from this error, because it used to be there even before Dumpling was enabled, like in the following run:
https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170123.01/workItem/Common.Tests/wilogs

About why Dumpling is trying to upload a file that doesn't exist, I'm still not sure. We do check to see if the file wasn't found, we don't proceed with uploading. I will look into this issue.
 
About `echo echo 0x3F > /proc/self/coredump_filter`, my guess is we probaly don't have permission in root to do this, but I need to check.

**Comment by schaabs** [comment]
@sepidehMS the issue with the command mentioned above is not a permissions issue you don't need root permissions to write to /proc/self/coredump_filter as the current process owns it.  The issue is the duplicated echo command.  My guess is that this line is intended to write out the intended command to the console and should instead be:

    echo "echo 0x3F > /proc/self/coredump_filter"

or if this line is actually intended to write to the file the extra echo would need to be removed.

    echo 0x3F > /proc/self/coredump_filter

**Comment by MattGal** [comment]
Chatted with @schaabs  and I think I understand this fairly well.  I think the simplest solution would be to nerf the word "echo" in lines today, either by removing it or replacing it with something that means nothing.  Alternatively, stripping all quotation from the lines to be echoed, then wrapping them with "" would also nerf the behavior.

**Comment by Jiayili1** [comment]
This issue is still repro on build 20170209.02, detail: https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170209.02/workItem/Common.Tests/wilogs

~~~
2017-02-09 06:39:58,402: INFO: proc(54): run_and_log_output: Output: echo uploading core to dumpling service
2017-02-09 06:39:58,403: INFO: proc(54): run_and_log_output: Output: python /home/helixbot/.dumpling/dumpling.py upload --dumppath --noprompt --triage full --displayname --properties STRESS_TESTID= --verbose
2017-02-09 06:39:58,403: INFO: proc(54): run_and_log_output: Output: else
2017-02-09 06:39:58,403: INFO: proc(54): run_and_log_output: Output: echo no coredump file was found
2017-02-09 06:39:58,403: INFO: proc(54): run_and_log_output: Output: fi
2017-02-09 06:39:58,403: INFO: proc(54): run_and_log_output: Output: fi
2017-02-09 06:39:58,403: INFO: proc(54): run_and_log_output: Output: return
2017-02-09 06:39:58,403: INFO: proc(54): run_and_log_output: Output: }
2017-02-09 06:39:58,403: INFO: proc(54): run_and_log_output: Output: /home/helixbot/dotnetbuild/work/427f7eb5-dcb9-4fae-a26a-b4bee16096cc/Payload/dotnet xunit.console.netcore.exe Common.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapp1.1tests -notrait category=OuterLoop -notrait category=failing -notrait category=nonlinuxtests
2017-02-09 06:39:58,403: INFO: proc(54): run_and_log_output: Output: CollectDumps 0 Common.Tests
2017-02-09 06:39:58,404: INFO: proc(54): run_and_log_output: Output: ~/dotnetbuild/work/427f7eb5-dcb9-4fae-a26a-b4bee16096cc/Work/5297eb58-1b27-4d52-b004-afd2b13f133c/Unzip ~/dotnetbuild/work/427f7eb5-dcb9-4fae-a26a-b4bee16096cc/Work/5297eb58-1b27-4d52-b004-afd2b13f133c/Unzip
2017-02-09 06:39:58,404: INFO: proc(54): run_and_log_output: Output: executing ulimit -c unlimited
2017-02-09 06:39:58,408: INFO: proc(54): run_and_log_output: Output: Failed to load /home/helixbot/dotnetbuild/work/427f7eb5-dcb9-4fae-a26a-b4bee16096cc/Payload/shared/Microsoft.NETCore.App/9.9.9/libhostpolicy.so, error: /lib64/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by /home/helixbot/dotnetbuild/work/427f7eb5-dcb9-4fae-a26a-b4bee16096cc/Payload/shared/Microsoft.NETCore.App/9.9.9/libhostpolicy.so)
2017-02-09 06:39:58,408: INFO: proc(54): run_and_log_output: Output: An error occurred while loading required library libhostpolicy.so from [/home/helixbot/dotnetbuild/work/427f7eb5-dcb9-4fae-a26a-b4bee16096cc/Payload/shared/Microsoft.NETCore.App/9.9.9]
2017-02-09 06:39:58,408: INFO: proc(54): run_and_log_output: Output: command exited with ExitCode: 130
2017-02-09 06:39:58,408: INFO: proc(54): run_and_log_output: Output: command failed, trying to collect dumps
2017-02-09 06:39:58,410: INFO: proc(54): run_and_log_output: Output: corefile:
2017-02-09 06:39:58,410: INFO: proc(54): run_and_log_output: Output: uploading core to dumpling service
2017-02-09 06:39:58,518: INFO: proc(54): run_and_log_output: Output: usage: dumpling.py upload [-h] [--verbose] [--squelch] [--noprompt]
2017-02-09 06:39:58,518: INFO: proc(54): run_and_log_output: Output:                           [--logpath LOGPATH] [--url URL]
2017-02-09 06:39:58,518: INFO: proc(54): run_and_log_output: Output:                           [--configpath CONFIGPATH] [--dumppath DUMPPATH]
2017-02-09 06:39:58,519: INFO: proc(54): run_and_log_output: Output:                           [--displayname DISPLAYNAME] [--user USER]
2017-02-09 06:39:58,519: INFO: proc(54): run_and_log_output: Output:                           [--triage {none,client,full}]
2017-02-09 06:39:58,519: INFO: proc(54): run_and_log_output: Output:                           [--incpaths [INCPATHS [INCPATHS ...]]]
2017-02-09 06:39:58,519: INFO: proc(54): run_and_log_output: Output:                           [--properties [key=value [key=value ...]]]
2017-02-09 06:39:58,519: INFO: proc(54): run_and_log_output: Output:                           [--propfile PROPFILE]
2017-02-09 06:39:58,519: INFO: proc(54): run_and_log_output: Output: dumpling.py upload: error: argument --dumppath: expected one argument
2017-02-09 06:39:58,528: INFO: proc(54): run_and_log_output: Output: ~/dotnetbuild/work/427f7eb5-dcb9-4fae-a26a-b4bee16096cc/Work/5297eb58-1b27-4d52-b004-afd2b13f133c/Unzip
2017-02-09 06:39:58,529: INFO: proc(54): run_and_log_output: Output: Finished running tests. End time=06:39:58. Return value was 130
~~~

**Comment by danmosemsft** [comment]
@mellinoe

**Comment by mellinoe** [comment]
Pretty sure this is going to be fixed by my latest buildtools PR, but we haven't started using that in corefx yet. https://github.com/dotnet/buildtools/pull/1328.

**Comment by Jiayili1** [comment]
This issue is still repro on Fedora, detail: https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170213.01/workItem/Common.Tests/wilogs

**Comment by danmosemsft** [comment]
@mellinoe can you put up a PR to update tools?

**Comment by mellinoe** [comment]
@danmosemsft I think it should be updated now (or rather, yesterday).

**Comment by Jiayili1** [comment]
Update this issue status, still repro on build 20170214.02, detail: https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170214.02/workItem/Common.Tests/wilogs

**Comment by Jiayili1** [comment]
Update: This issue is not repro on build 20170215.01.

**Comment by danmosemsft** [comment]
great thanks

**Comment by MattGal** [comment]
@danmosemsft , I don't think this means this bug is fixed, simply that we made it so the tests don't crash.  If we care about dumpling working in this scenario, it still doesnt.  I will get Fedora 23 actually cleaned up today.

Just to make sure you understand why: 
- I was attempting to disable Fedora 23 runs entirely since 23 is EOL, and we had been building for 24 and sending those to the 23 queue, causing a crash.
- I merged a PR which I thought acheived this yesterday ( https://github.com/dotnet/corefx/pull/16151 )
- I did not notice that there actually also were Fedora 23 builds still present.

What happened here was that since work items with the same name were put into the same queue, a "last wins" thing happened, resulting in this bug.
